from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "follow_control"

# Create generator
gen = ParameterGenerator()

# Add parameters to the generator
gen.add("joint_vel_ratio", double_t, 0, "joint 速度与位置差的比例", 10, 0, 20)
gen.add("joint_pid_Kp", double_t, 0, "joint p", 0.25, 0, 1)
gen.add("joint_pid_Ki", double_t, 0, "joint i", 0.01, 0, 0.1)
gen.add("joint_pid_Kd", double_t, 0, "joint d", 0, 0, 1.5)
gen.add("joint_current_limit", double_t, 0, "joint 最大电流限制", 300, 0, 1200)

many_enum = gen.enum([gen.const("joint1",int_t,0,"joint1 number"),
                      gen.const("joint2",int_t,1,"joint2 number"),
                      gen.const("joint3",int_t,2,"joint3 number"),
                      gen.const("joint4",int_t,3,"joint4 number"),
                      gen.const("joint5",int_t,4,"joint5 number"),
                      gen.const("joint6",int_t,5,"joint6 number"),
                      gen.const("joint7",int_t,6,"joint7 number")
                     ],"joint serial number")

gen.add("joint",int_t,0,"关节序号",0,0,2, edit_method=many_enum)

# Generate intermediate file and exit
exit(gen.generate(PACKAGE,"dynamic_param","dynamic_param"))